client/engine/engine.go

```diff
@@ -126,11 +126,12 @@ func (e *Engine) ToApi() <-chan EngineEvent {
 }
 
 func (e *Engine) Close() error {
+	if err := e.msg.Close(); err != nil {
```

Comment:
Nit but it might be nice to document why we're closing things in a particular order.

Reply:
I am actually not convinced that this order is the correct order. This was one of the early, experimental changes when I knew less about how an engine is shut down.